### ISSUE TITLE
tools: make combine logic align with FIT Activity CLI

### ISSUE THREAD

**Issue by muktihari** [openivity/openivity.github.io#114]
Currently, the logic on combining activities is way behind FIT Activity CLI, means there are bugs that have been fixed on FIT Activity CLI but still lying around in this project. Some of the bugs that I could list for now are:

1. Handling values accumulation across all messages hasn't been implemented (currently only Record's Distance is being accumulated). 
2.  Handling Split Summary aggregation (currently we silently discard this message) 

These bugs are causing data loss. 

I believe that data is precious, we should keep the loss to minimum. Moreover, CLI is not for everyone, having GUI version, that is also reliable, will accommodate more people. To tackle this, I plan to implement various changes to update the implementation, #112 and #113 are the starting points.

**Comment by muktihari** [comment]
I am thinking about using the [FIT Activity CLI](https://github.com/muktihari/fit/blob/master/cmd/fitactivity/README.md) as tools' engine, so we can maintain only one single source code. I think [FIT Activity CLI](https://github.com/muktihari/fit/blob/master/cmd/fitactivity/README.md) is now battle-tested again various FIT files. The openivity tools' logic has many flaws, we can't restructure the FIT protocol messages as the original order as we group it together per message type, such as Sessions, Records, etc. And if we combine two FIT files that has developer data, I can't guarantee that the resulting file will be valid as the **developer_data** and **field_description** messages may overlap. While [FIT Activity CLI](https://github.com/muktihari/fit/blob/master/cmd/fitactivity/README.md)maintains the order as it was.

For GPX and TCX we can convert it into FIT first before we can use the engine. Then convert it again to GPX or TCX if users want to export in those format, I think the conversion is far easier since we already have the module, and it should not take long to process, even it it takes a bit longer, so be it. FIT file is the first class citizen here as it offer more advanced features.

Now that it has been 1 year, I feel like maintaining two source code is hard. At the beginning, I was thinking that having two implementation will help finding the finding the best way to do the logic, and it does help, [FIT Activity CLI](https://github.com/muktihari/fit/blob/master/cmd/fitactivity/README.md) implementation wins here.

One and only missing part of [FIT Activity CLI](https://github.com/muktihari/fit/blob/master/cmd/fitactivity/README.md) is that I haven't implemented the Trim Tracks logic yet since I rarely use that feature, but it should not be harder than fixing the current bugs and maintaining two source code for tools.

cc: @raditzlawliet

**Comment by muktihari** [comment]
[FIT Activity CLI](https://github.com/muktihari/fit/blob/master/cmd/fitactivity/README.md) is modular and it has proper unit test. And it's far easier to read.

**Comment by raditzlawliet** [comment]
As I know currently Openivity is for user-oriented who can use with minimum UI. And use Fit SDK Go as library to process the data. 
Since Openivity will use .fit file up to latest version, I Agree that 2 points you mention can be included.

For the tools, Did you mean you want to move current Fit processing from Openivity (Which currently use only FIT SDK) to FIT SFK Library with **FIT Activity CLI** (cmiiw).

**Comment by muktihari** [comment]
> For the tools, Did you mean you want to move current Fit processing from Openivity (Which currently use only FIT SDK) to FIT SFK Library with FIT Activity CLI (cmiiw).

Yes, correct. FIT Activity CLI is actually not the part of the SDK itself but it's a supplementary tools that comes with the SDK. If there is an update or new feature implemented, openivity only need to bump the version and wire the new feature as needed.

This way, we can focus on developing openivity as the presentation layer.

**Comment by muktihari** [comment]
I'll postponed this issue since it's look like I don't have time to refactor, I'll do the patches for common bug first, and leave the edge-case bugs open, I think regular people don't need that kind of data precision.
